lib/GameObject.ts

```diff
@@ -123,33 +123,7 @@ class GameObject {
 
 	public removeAllReferencesToGameObject(gameObject: GameObject) {
 		for (let key in this as any) {
-			if (this[key] != null) {
-				// remove gameObject from arrays:
-				// like if this.cars[3] === gameObject
-				if (this[key].constructor === Array) {
-					for (let i=this[key].length-1; i>=0; i--) {
-						if (this[key][i] === gameObject) {
-							this[key][i] = null;
-						}
-					}
-				}
-
-				// remove direct references to gameObject:
-				// like if this.car === gameObject
-				else if (this[key] === gameObject) {
-					this[key] = null;
-				}
-
-				// remove object references:
-				// like if this.car.driver === gameObject
-				else if (typeof this[key] === 'object') {
-					for (let objKey in this[key]) {
-						if (this[key][objKey] === gameObject) {
-							this[key][objKey] = null;
-						}
-					}
-				}
-			}
+			delete this[key];
```

Comment:
This doesn't accomplish the same functionality, it doesn't do a shallow search for references to the object.

You've change it  from "remove all references to game object" to "remove all references in game object"

Reply:
a.list = [b, c]

removeAllReferencesToGameObject(b)

a.list will still have [b,c]